anonlink/similarities/dice.cpp

```diff
@@ -516,12 +524,16 @@ extern "C"
             max_popcnt_delta = calculate_max_difference(count_one, threshold);
         }
 
+        double dynamic_threshold = threshold;
         auto push_score = [&](double score, int idx) {
-            if (score >= threshold) {
+            if (score >= dynamic_threshold) {
                 top_k_scores.push(Node(idx, score));
                 if (top_k_scores.size() > k) {
                     // Popping the top element is O(log(k))!
+                    temp_node = top_k_scores.top();
                     top_k_scores.pop();
+                    // threshold can now be raised
+                    dynamic_threshold = temp_node.score;
```

Comment:
wait, the `top_k_scores` is a priority queue where the lowest score has the highest priority? Thank god that's so well documented, otherwise this would be quite confusing...

Reply:
Added a comment where the priority queue is created